src/application.ts

```diff
@@ -33,10 +33,12 @@ export default class Application {
   }
 
   async executeOperations(ops: Operation[]): Promise<OperationResponse[]> {
+    let applicationInstance = new ApplicationInstance(this);
+
     return await this.createTransaction(
       ops
         .map(async op => {
-          const processor = await this.processorFor(op.ref.type);
+          const processor = await this.processorFor(op.ref.type, applicationInstance);
```

Comment:
should processorFor move to the instance so we don't have to pass it?

Reply:
Yeah, we can change that.